### ISSUE TITLE
Epoch shows less then actual data

### ISSUE THREAD

**Issue by Night1099** [giuvecchio/matfuse-sd#10]
my Epoch shows 490/25003

But i have 1.5million in dataset and script sees and says that

1 gpu, 2 batch size, 2 gradient accumulation

Shouldnt it be x/375000

**Comment by Night1099** [comment]
i just found the line
```
subsample_size = min(50000, len(self.datasets["train"]))
```

so it is purposefull and will select a random number for each batch of epoch
